### PR TITLE
Components: Add `info` prop support to MenuItem, FeatureToggle

### LINKED ISSUE #10340: Add Option for Descriptions within Menu Items
Let's add the ability to display a lengthier description under Menu Items, similar to how the post visibility works.

image

It would be useful to use in cases like the focus modes where it's hard for a single title to be self-explanatory.

image

### PR DESCRIPTION
**Pull request by aduth**
Closes #10340 

This pull request seeks to add a new `info` prop to the `MenuItem` and `FeatureToggle` components.

image

**Testing Instructions:**

Verify an extended description is shown for the More Menu Unified Toolbar button, as in the screenshot above:

Notable variations:

- Checked, unchecked
- Cross-browser

Ensure unit tests pass:

```
npm run test-unit
```

### CODE REVIEW

File: packages/components/src/menu-item/style.scss [hunk 1]
```diff
@@ -24,13 +24,16 @@
 	&:focus:not(:disabled):not([aria-disabled="true"]) {
 		@include menu-style__focus;
 	}
+}
 
-	// Don't wrap text until viewport is beyond the mobile breakpoint.
```
Comment (aduth) [inline packages/components/src/menu-item/style.scss]: This change directly undoes the intention of #8756.

Guess it depends if we'd rather:

Wrapped|Unwrapped
---|---
Wrapped|Unwrapped

Also, the comment is misleading, where the previous behavior had the opposite effect.

cc @jasmussen @tofumatt

File: test/e2e/specs/links.test.js [hunk 2]
```diff
@@ -192,17 +192,14 @@ describe( 'Links', () => {
...
-		const button = ( await page.$x( "//button[contains(text(), 'Unified Toolbar')]" ) )[ 0 ];
-		const buttonClassNameProperty = await button.getProperty( 'className' );
-		const buttonClassName = await buttonClassNameProperty.jsonValue();
-		const isSelected = buttonClassName.indexOf( 'is-selected' ) !== -1;
-		if ( isSelected !== b ) {
-			await button.click();
-		} else {
-			await page.click( '.edit-post-more-menu button' );
-		}
+	const toggleFixedToolbar = async ( isFixed ) => {
```
Comment (aduth) [inline test/e2e/specs/links.test.js]: cc @noisysocks re https://github.com/WordPress/gutenberg/pull/8807#discussion_r226692130

File: packages/components/src/icon-button/index.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ class IconButton extends Component {
 		);
 
 		let element = (
-			<Button { ...additionalProps } aria-label={ label } className={ classes }>
+			<Button aria-label={ label } { ...additionalProps } className={ classes }>
```
Comment (youknowriad) [inline packages/components/src/icon-button/index.js]: Can't you just use the `label` prop?

File: packages/components/src/menu-item/README.md [hunk 4]
```diff
@@ -18,3 +20,48 @@ const MyMenuItem = withState( {
...
+- Required: No
+
+Refer to documentation for [IconButton's `icon` prop](../icon-button/README.md#icon).
+
+### `shortcut`
+
+- Type: `string`
+- Required: No
+
+Refer to documentation for [Shortcut's `shortcut` prop](../shortcut/README.md#shortcut).
```
Comment (youknowriad) [inline packages/components/src/menu-item/README.md]: Thanks for the detailed README here 👍

File: packages/components/src/icon-button/index.js (hunk 3 above)
Comment (aduth) [reply packages/components/src/icon-button/index.js]: Not sure what you mean by "use". `Button` doesn't have a `label` prop.

**Comment by youknowriad** [comment]
Design questions

 - Can't we leave the shortcuts there somehow?
 - Can we reduce the "opacity" or change a little bit the color of the description (to separate it more clearly from the title) (I think opacity: 0.84; is the minimum opacity to be accessible)

**Comment by youknowriad** [comment]
Code wise, this look good 👍

**Comment by aduth** [comment]
>Can't we leave the shortcuts there somehow?

Which shortcuts?

>Can we reduce the "opacity" or change a little bit the color of the description

We already have two other variations of a descriptive subtext, one in the Post Visibility menu, the other of sidebar Controls (e.g. Drop Cap description). As implemented here, I've inherited precisely the style used for the sidebar controls (as proposed at https://github.com/WordPress/gutenberg/issues/10340#issuecomment-427876823 and seconded at https://github.com/WordPress/gutenberg/issues/10340#issuecomment-428325642).

Now, that being said, I kinda agree with you 😄 But I didn't want to go ahead and add a _third_ variation of what amounts to the same usage.

**Comment by youknowriad** [comment]
@aduth I was confused, just noticed these modes don't have shortcuts, I thought they had.

**Review by youknowriad — APPROVED** [review]
LGTM 👍 (I'd appreciate a design check first though)

File: packages/components/src/menu-item/style.scss (hunk 1 above)
Comment (mtias) [reply packages/components/src/menu-item/style.scss]: @alexislloyd thoughts on this?

**Comment by youknowriad** [comment]
cc @alexislloyd in case you have some thoughts on the design questions raised here

  - https://github.com/WordPress/gutenberg/pull/10802#issuecomment-431416369 (second point)
 - https://github.com/WordPress/gutenberg/pull/10802#discussion_r226695962

File: packages/components/src/menu-item/style.scss (hunk 1 above)
Comment (aduth) [reply packages/components/src/menu-item/style.scss]: Maybe worth noting currently there _is_ a max-width of the menu (480px), after which the text simply becomes clipped.

image

Comment (alexislloyd) [reply packages/components/src/menu-item/style.scss]: I think wrapped is better here, as long as we're careful not to have the descriptions go over two lines with standard font sizes.

Comment (alexislloyd) [reply packages/components/src/menu-item/style.scss]: I'm also ok with editing down the first description to "Access all tools in one place"

Comment (aduth) [reply packages/components/src/menu-item/style.scss]: We also have the option to enforce a max-width (with wrapping) on the description specifically, allowing both the main label and shortcut text grow unrestrained.

**Comment by youknowriad** [comment]
screen shot 2018-10-19 at 19 37 01

Updated the PR:

 - I made the menu links a bit darker, the info lighter and decreased the font-size for the info by 1px (2px felt too small)

Thoughts?

**Pull request merged**